api/v1/observer/observer.proto

```diff
@@ -105,6 +106,11 @@ message GetFlowsRequest {
 
     // Until this time for returned flows. Incompatible with `number`.
     google.protobuf.Timestamp until = 8;
+
+    // FieldMask allows clients to limit flow's fields that will be returned.
+    // For example, {paths: ["source.id", "destination.id"]} will return flows
+    // with only these two fields set.
+    google.protobuf.FieldMask field_mask = 10;
```

Comment:
One of the things I'd like to suggest is to first try this out without committing to this particular API in v1 forever after merging this PR and including it in the upcoming release.

Most of these `experimental APIs` are only enforced through documentation, but in this particular case I'd like to go one step further. I think we should include a `message Experimental` inside `message GetFlowsRequest` so we can try out features, gather feedback, and then come back to improve without the burdens of API compatibility.

If we decide in the future that actually this string based field mask is what we'd like to proceed with we can promote from `GetFlowsRequest.Experimental` into `GetFlowsRequest`.

@cilium/sig-hubble any thoughts on this?

Reply:
@glibsm sounds good to me.

Reply:
Done. Github messes up diffs on rebases, so let me know once you are ok with the current state of the changes